Notely/lib/screens/authenticate/register.dart

```diff
@@ -0,0 +1,90 @@
+class _RegisterState extends State<Register> {
+  final AuthService _auth = AuthService();
+  final _formKey = GlobalKey<FormState>();
+
+  String email = '';
+  String password = '';
+  String error = '';
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        backgroundColor: Colors.white,
+        appBar: AppBar(
+          backgroundColor: Colors.tealAccent.shade400,
+          elevation: 0.0,
+          title: Text('Sign up to Notely',),
+          actions: <Widget>[
+            FlatButton.icon(
+                icon: Icon(Icons.person, color: Colors.white,),
+                label: Text('Sign In', style: TextStyle(color: Colors.white)),
+                onPressed: () {
+                  widget.toggleView();
+                })
+          ],
+        ),
+        body: Container(
+          padding: EdgeInsets.symmetric(vertical: 20.0, horizontal: 50.0),
+          child: Form(
+              key: _formKey,
+              child: Column(
+                children: <Widget>[
+                  SizedBox(height: 20.0),
+                  TextFormField(
+                    decoration: const InputDecoration(
+                      icon: Icon(Icons.person, ),
+                      hintText: 'Enter a email',
+                      labelText: 'Email',
+                    ),
+                    validator: (val) => val.isEmpty ? 'Enter an email' : null,
+                    onChanged: (val) {
+                      setState(() => email = val);
+                    },
+                  ),
+                  SizedBox(height: 20.0),
+                  TextFormField(
+                    decoration: const InputDecoration(
+                      icon: Icon(Icons.person),
+                      hintText: 'Enter a password',
+                      labelText: 'Password',
+                    ),
+                    obscureText: true,
+                    validator: (val) => val.length < 6 ? 'Enter a password over 6 characters long' : null,
+                    onChanged: (val) {
+                      setState(() => password = val);
+                    },
+                  ),
+                  SizedBox(height: 20.0),
+                  RaisedButton(
+                      color: Colors.tealAccent.shade400,
+                      child: Text('Register', style: TextStyle(color: Colors.white,)),
+                      onPressed: () async {
+                        if (_formKey.currentState.validate()) {
+                          dynamic result = await _auth.registerWithEmailAndPassword(email, password);
+                          if (result == null) {
+                            setState(() => error = "Please supply a valid email");
+                          }
+                        }
+                      }),
+                  SizedBox(height: 12.0),
+                  Text(
+                    error,
+                    style: TextStyle(color: Colors.red, fontSize: 14.0),
+                  )
+                ],
```

Comment:
The code is written very well, however, it would be nice if it were commented a little more.